config/initializers/app_version.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+if Rails.env.development?
+  if File.directory? File.expand_path(".git")
+    system("git fetch #{Rails.application.config.upstream_repo}")
+    git_sha = `git rev-parse --short=8 HEAD`
```

Comment:
```suggestion
    git_sha = `git rev-parse HEAD`
```

shortening the SHA is left to the view now, so it's only done when needed